.github/workflows/main.yml

```diff
@@ -25,6 +25,9 @@ jobs:
       - name: Build
         run: ./scripts/build.sh
 
+      - name: Update API Description
+        run: ./scripts/update_api_description.sh
```

Comment:
This is no good and defeats the purpose of testing the API description.
You are generating it here and then just testing it against the same generation in the unittest ...
Nowhere did you actually validate that the test is correct.

The whole idea is that we need a person to verify that the API description is correct.
This means that you need an API file to always live in the repo.

The general workflow for someone working on an endpoint would be:
 - Work on endpoint.
 - Run tests, which fail because they forgot to update the API description.
 - Look at changes to description (could be done directly through the unit test, since they will see a diff there).
 - User accepts changes to to the API (or finds an issue and goes back and fixes it).
 - User runs `./scripts/update_api_description.sh` to codify their changes.
 - Run tests again (which should pass).
 - Done.